test/tests/functional/pbs_reservations.py

```diff
@@ -1123,3 +1123,128 @@ def test_reservations_with_expired_subjobs(self):
         # Check status of the parent-job using qstat -fx once reservation ends
         self.server.expect(JOB, {'job_state': 'F', 'substate': '91'},
                            extend='x', id=jid)
+
+    def common_config(self):
+        """
+        This function contains common steps for test
+        "test_ASAP_resv_with_multinode_job_array" and
+        "test_standing_resv_with_multinode_job_array"
+        """
+        vn_attrs = {ATTR_rescavail + '.ncpus': 4}
+        self.server.create_vnodes("vnode1", vn_attrs, 1,
+                                  self.mom, fname="vnodedef1")
+        self.server.create_vnodes("vnode2", vn_attrs, 2,
+                                  self.mom, additive=True, fname="vnodedef2")
+        self.server.manager(MGR_CMD_SET, SERVER,
+                            {'job_history_enable': 'True'})
+
+    def test_ASAP_resv_with_multivnode_job_array(self):
+        """
+        Test multivnode job array with ASAP reservation. Also
+        verify that the the 2 ASAP reservations with same starting time
+        doesn't crashes PBS daemons
+        """
+        self.common_config()
+        attrs = {ATTR_J: '1-5',
+                 'Resource_List.select': '2:ncpus=1',
+                 'Resource_List.walltime': '15',
+                 'Resource_List.place': 'vscatter'}
+        j = Job(PBSROOT_USER)
+        j.set_sleep_time(15)
+        j.set_attributes(attrs)
+        jid = self.server.submit(j)
+        subjid = []
+        for i in range(1, 6):
+            subjid.append(j.create_subjob_id(jid, i))
+        self.server.expect(JOB, {'job_state=R': 4}, count=True,
+                           extend='t', id=jid)
+        self.server.expect(JOB, {'job_state=Q': 1}, count=True,
+                           extend='t', id=jid)
+        now = int(time.time())
+        rid1 = self.submit_asap_reservation(PBSROOT_USER, jid)
+        rid1_q = rid1.split('.')[0]
+        j1 = Job(PBSROOT_USER)
+        j1.set_sleep_time(15)
+        j1.set_attributes(attrs)
+        jid2 = self.server.submit(j1)
```

Comment:
Please write a comment about why you are submitting second array job and how this job will help you get the same start time as the previous ASAP reservation